Various small documentation Fixes

TcT2k opened the pull request:
- iOS build instructions
- wxNotificationMessage event docs
- AUI native art

TcT2k committed:
Add architecture i386 to iOS build instructions.

The 64-bit intel build currently fails and it is really not necessary.

TcT2k committed:
Update AUI docs on native art providers.

Include information about native toolbar and tab art in wxMSW.

TcT2k committed:
Add wx prefix in wxNotificationMessage event docs.

MaartenBent commented:
[wxListCtrl::EnableCheckboxes](http://docs.wxwidgets.org/trunk/classwx_list_ctrl.html#a6e89dd0e441f2f39933193f301ebcfda) is not MSW-only anymore. Can you remove this too?

Maybe add `[ci skip]` to the commit message so appveyor builds do not pile up.

TcT2k committed:
Update wxListCtrl::EnableCheckboxes documentation.

Method is no longer MSW only.

TcT2k committed:
Drop Mac prefix on remaining instances of OS X in docs.

This should have been part of df33c0c49162f7218e14f525ce626501b6b583a6 but somehow where missed.

TcT2k committed:
Remove wxOSX/Carbon only wxFileName methods from docs.

The carbon only implementation was removed in 5ba67c67e47d069f65d648ab16dfc505c5400bfc.
